centinel/primitives/dnslib.py

```diff
@@ -94,12 +94,8 @@ def lookup_domains(self):
         """
         for nameserver in self.nameservers:
             for domain in self.domains:
-                escape_count = 0
                 while threading.active_count() > self.max_threads:
                     time.sleep(1)
-                    escape_count += 1
-                    if escape_count > 100:
```

Comment:
I have added the timeout, but found a bigger problem with the multithreaded `traceroute` primitive. If one thread ends prematurely as determined by the callback function in your `Command` class, all other threads will be killed as well (issue #130). It might have something to do with the way you've implemented `Command` (e.g. using a thread, which is within another thread).

This means that when running `traceroute` concurrently and one of the domains doesn't exist, all of the other traceroutes that are running at the same time will be killed.
I don't have a solution for this for now. I've tried removing `self.kill_switch()` from output processing callback function of the `traceroute` primitive, but it did not help resolve the issue.